### PR TITLE
cortexm_common: don't compare with r13/sp in assembly, fix #3664

### LINKED ISSUE #3664: cortexm_common: use of r13 is deprecated
Since the merge of #3333 I get the following warning:

```
$ BOARD=iot-lab_M3 IOTLAB_NODES=2 make -C examples/gnrc_networking/ all
make: Entering directory `/home/martine/Repositories/RIOT-OS/RIOT/examples/gnrc_networking'
Building application "gnrc_networking" for "iot-lab_M3" with MCU "stm32f1".

"make" -C /home/martine/Repositories/RIOT-OS/RIOT/boards/iot-lab_M3
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/core
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32f1
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/cortexm_common
/tmp/ccwJhJJ3.s: Assembler messages:
/tmp/ccwJhJJ3.s:211: Warning: use of r13 is deprecated
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32f1/periph
[…]
```

### PR DESCRIPTION
**Pull request by daniel-k**
Should fix #3664.

Please test this, as I have no hardware at hand atm.

### CODE REVIEW

File: cpu/cortexm_common/vectors_cortexm.c [hunk 1]
```diff
@@ -134,7 +134,8 @@ void hard_fault_default(void)
         "mrs r0, psp                        \n" /*   r0 = psp                 */
         " out:                              \n" /* }                          */
         "mov r1, #0                         \n" /* corrupted = false          */
-        "cmp r0, sp                         \n" /* If msp is active stack-    */
+        "mov r2, sp                         \n" /* Cmp with SP is deprecated  */
+        "cmp r0, r2                         \n" /* If msp is active stack-    */
```
Comment (thomaseichinger) [inline cpu/cortexm_common/vectors_cortexm.c]: Couldn't we use `cmp sp, r0`? `sp` is only deprecated as second operant for `cmp`. Compare https://github.com/RIOT-OS/RIOT/issues/3664#issuecomment-132743467

**Comment by PeterKietzmann** [comment]
I can confirm that this PR fixes the warning described in #3664 but I'm not in position of ACKing it. Would be nice to have this merged soon

**Comment by gebart** [comment]
Untested ACK. I'm on my mobile and can't test. Peter, did you test the actual code as well?
This code can be tested by adding something like `*((volatile uint32_t*)0) = 123;` inside the hello world example.

**Comment by PeterKietzmann** [comment]
What would be the expected outcome with or without this PR @gebart ? Independent from your proposed test, I see the warning described in #3664 without this PR, and it disappears after merging the PR.

**Comment by gebart** [comment]
@PeterKietzmann the result would be that a hard fault will occur and the debugging information is printed to stdout, just to make sure that the code still works after silencing the warning.

**Comment by PeterKietzmann** [comment]
It seems so:
```
2015-08-20 11:02:05,013 - INFO # kernel_init(): jumping into first task...
2015-08-20 11:02:05,014 - INFO # Hello World!
2015-08-20 11:02:05,014 - INFO # 
2015-08-20 11:02:05,015 - INFO # Context before hardfault:
2015-08-20 11:02:05,015 - INFO # r0:  0xa
2015-08-20 11:02:05,016 - INFO # r1:  0x8002757
2015-08-20 11:02:05,016 - INFO # r2:  0x80000000
2015-08-20 11:02:05,016 - INFO # r3:  0x7b
2015-08-20 11:02:05,016 - INFO # r12: 0x0
2015-08-20 11:02:05,016 - INFO # lr : 0x8000647
2015-08-20 11:02:05,017 - INFO # pc : 0x8001090
2015-08-20 11:02:05,017 - INFO # psr: 0x21000000
2015-08-20 11:02:05,017 - INFO # 
2015-08-20 11:02:05,017 - INFO # *** RIOT kernel panic
2015-08-20 11:02:05,017 - INFO # HARD FAULT HANDLER
2015-08-20 11:02:05,018 - INFO # *** halted.
2015-08-20 11:02:05,018 - INFO # 
```

File: cpu/cortexm_common/vectors_cortexm.c (hunk 1 above)
Comment (gebart) [inline cpu/cortexm_common/vectors_cortexm.c]: merging this as-is right now to fix errors in Travis, I will post a follow-up with some added debugging features for hard faults

**Comment by gebart** [comment]
ACK, waiting for Travis.

**Pull request merged**